File: Java/src/scitrader.marketdataserver/src/main/java/com/scitrader/marketdataserver/datastore/aggregators/TickAggregatorBase.java
package com.scitrader.marketdataserver.datastore.aggregators;

import com.mongodb.client.FindIterable;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.model.Filters;
import com.scitrader.marketdataserver.exchange.bitmex.Tick;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.bson.conversions.Bson;
import org.joda.time.DateTime;
import org.bson.Document;

public abstract class TickAggregatorBase {

  protected final Logger Log = LogManager.getLogger(this.getClass());

  FindIterable<Document> fetchTicks(MongoCollection<Document> ticksCollection, DateTime from, DateTime to) {
    long startDate = from.getMillis();
    long endDate = to.getMillis();

    Log.info(String.format("Filtering by start='%d', end='%d", startDate, endDate));
    Bson filter = Filters.and(Filters.gte("Time", startDate), Filters.lte("Time", endDate));
    FindIterable<Document> foundTicks = ticksCollection.find(filter);
    return foundTicks;
  }
}

File: Kotlin/hello-kotlin-world/src/main/java/com/scitrader/marketdataserver/datastore/aggregators/TickAggregatorBase.kt
package com.scitrader.marketdataserver.datastore.aggregators

import com.mongodb.client.FindIterable
import com.mongodb.client.MongoCollection
import com.mongodb.client.model.Filters
import org.apache.logging.log4j.LogManager
import org.bson.Document
import org.joda.time.DateTime

abstract class TickAggregatorBase : ITickAggregator {

    protected val Log = LogManager.getLogger(this.javaClass)

    internal fun fetchTicks(ticksCollection: MongoCollection<Document>, from: DateTime, to: DateTime): FindIterable<Document> {
        val startDate = from.millis
        val endDate = to.millis

        Log.info(String.format("Filtering by start='%d', end='%d", startDate, endDate))
        val filter = Filters.and(Filters.gte("Time", startDate), Filters.lte("Time", endDate))
        return ticksCollection.find(filter)
    }
}

